src/shapes/Object/Object.ts

```diff
@@ -895,20 +903,24 @@ export class FabricObject<
    * @param {Object} object
    */
   _removeDefaultValues(object: Record<string, any>) {
-    const prototype = classRegistry.getClass(object.type).prototype;
+    const topLevelClass = classRegistry.getClass(object.type);
```

Comment:
```suggestion
    const topLevelClass = this.constructor;
```

Reply:
it get confused i think when you try to access prototype from this.constructor, i need to see if it gives me back the `Function` prototype instead of the class one. i ll give it a shot

Reply:
what about this?